tests/transformers/sampler/test_sampler.py

```diff
@@ -475,3 +535,121 @@ def test_random_sampling(
+            "include_sampler": True,
+            "return_pdfs": False,
+            "max_top_k_ids": 1024,
+            "include_guided_decoding": True,
+        },
+        **additional_configs,
+    )
+    model_w_sampler_wo_guided_decoding = qeff_class.from_pretrained(
+        model,
+        continuous_batching=True,
+        qaic_config={
+            "include_sampler": True,
+            "return_pdfs": False,
+            "max_top_k_ids": 1024,
+        },
+        **additional_configs,
+    )
+    model_w_sampler_w_guided_decoding.compile(
+        prefill_seq_len=prefill_seq_len,
+        ctx_len=ctx_len,
+        full_batch_size=full_batch_size,
+        num_devices=1,
+        num_cores=16,
+        num_speculative_tokens=spec_length,
+        mxint8_kv_cache=True,
+        mxfp6_matmul=True,
+    )
+    model_w_sampler_wo_guided_decoding.compile(
+        prefill_seq_len=prefill_seq_len,
+        ctx_len=ctx_len,
+        full_batch_size=full_batch_size,
+        num_devices=1,
+        num_cores=16,
+        num_speculative_tokens=spec_length,
+        mxint8_kv_cache=True,
+        mxfp6_matmul=True,
+    )
+
+    # Generate texts from prompts
+    tokenizer = load_hf_tokenizer(pretrained_model_name_or_path=model)
+    np.random.seed(0)
+    sampling_params = {
+        "repetition_penalties": np.array(1.0, dtype=np.float32).repeat(full_batch_size).reshape(-1, 1),
+        "presence_penalties": np.array(0.0, dtype=np.float32).repeat(full_batch_size).reshape(-1, 1),
+        # "frequency_penalties": np.array(0.0, dtype=np.float32).repeat(full_batch_size).reshape(-1, 1),
+        "temperatures": np.array(0.0, dtype=np.float32).repeat(full_batch_size).reshape(-1, 1),
+        "top_ks": np.array(1024, dtype=np.int32).repeat(full_batch_size).reshape(-1, 1),
+        "top_ps": np.array(1.0, dtype=np.float32).repeat(full_batch_size).reshape(-1, 1),
+        "min_ps": np.array(0.0, dtype=np.float32).repeat(full_batch_size).reshape(-1, 1),
+        "random_numbers": np.zeros((full_batch_size, 1024), dtype=np.float32),
+    }
+    model_w_sampler_w_guided_decoding_exec_info = model_w_sampler_w_guided_decoding.generate(
+        tokenizer=tokenizer,
+        prompts=prompts,
+        generation_len=generation_len,
+        include_sampler=True,
+        return_pdfs=False,
+        include_guided_decoding=True,
+        sampling_params={
+            **sampling_params,
+            **{
+                "token_bitmasks": np.tile(
+                    np.random.choice([True, False], size=(model_w_sampler_w_guided_decoding.model.config.vocab_size,)),
+                    (full_batch_size, 1),
+                )
+            },
+        },
+    )
+    model_w_sampler_wo_guided_decoding_exec_info = model_w_sampler_wo_guided_decoding.generate(
+        tokenizer=tokenizer,
+        prompts=prompts,
+        generation_len=generation_len,
+        include_sampler=True,
+        return_pdfs=False,
+        sampling_params=sampling_params,
```

Comment:
Same as the above comment, additional_params need to be added here.
```suggestion
        sampling_params=sampling_params,
        **additional_params,
```